lib/membrane_rtsp/parser.ex

```diff
@@ -0,0 +1,112 @@
+defmodule Membrane.RTSP.Parser do
+  @moduledoc false
+
+  import NimbleParsec
+
+  @methods [
+    "DESCRIBE",
+    "ANNOUNCE",
+    "GET_PARAMETER",
+    "OPTIONS",
+    "PAUSE",
+    "PLAY",
+    "RECORD",
+    "REDIRECT",
+    "SETUP",
+    "SET_PARAMETER",
+    "TEARDOWN"
+  ]
+
+  alpha_num = ascii_string([?0..?9, ?a..?z, ?A..?Z, ?-, ?_], min: 1)
+  space = ascii_string([?\s, ?\t, ?\f], min: 1)
+  delimeter = choice(Enum.map(["\r\n", "\r", "\n"], &string/1))
+
+  ignored_space = ignore(space)
+  ignored_delimeter = ignore(delimeter)
+
+  methods = choice(Enum.map(@methods, &string/1))
+
+  header =
+    alpha_num
+    |> ignore(string(":"))
+    |> concat(ignored_space)
+    |> utf8_string([{:not, ?\r}, {:not, ?\n}], min: 1)
+    |> concat(ignored_delimeter)
+
+  request_line =
+    methods
+    |> concat(ignored_space)
+    |> utf8_string([{:not, ?\s}], min: 1)
+    |> concat(ignored_space)
+    |> ignore(string("RTSP/1.0"))
+    |> concat(ignored_delimeter)
+
+  headers = header |> times(min: 0) |> post_traverse(:parse_headers)
+
+  defparsec :parse_request,
+            request_line
+            |> optional(headers)
+            |> concat(ignored_delimeter)
+            |> optional(utf8_string([], min: 1))
+            |> eos()
+
+  # Parse Transport header
+  transport_param_value = utf8_string([{:not, ?\0..?\s}, {:not, ?;}], min: 1)
+
+  transport_dest_param =
+    string("destination") |> optional(string("=") |> concat(transport_param_value))
```

Comment:
This is confusing. Breaking it into two lines is a bit better IMO

```suggestion
  transport_dest_param =
    string("destination")
    |> optional(string("=") |> concat(transport_param_value))
```

Reply:
One more thing, the RFC seems to specify an analogous parameter `source` as well, so we should add that